core/src/androidMain/kotlin/Scanner.kt

```diff
@@ -60,22 +61,36 @@ public class AndroidScanner internal constructor(
             }
         }
 
-        val scanFilter = filterServices
+        val manufacturerDataScanFilters = manufacturerDataFilters
+            ?.map {
+                ScanFilter.Builder()
+                    .setManufacturerData(
+                        it.manufacturerId,
+                        it.manufacturerData,
+                        it.manufacturerDataMask
+                    )
+                    .build()
+            }
+        val serviceScanFilters = filterServices
             ?.map { ScanFilter.Builder().setServiceUuid(ParcelUuid(it)).build() }
-            ?.toList()
+        val scanFilters = (manufacturerDataScanFilters ?: emptyList())
+            .plus(serviceScanFilters ?: emptyList())
```

Comment:
nitpick

```suggestion
        val scanFilters = manufacturerDataScanFilters.orEmpty() + serviceScanFilters.orEmpty()
```

Reply:
Ah, I didn't know that `orEmpty()` function existed! Agreed, I like it here.